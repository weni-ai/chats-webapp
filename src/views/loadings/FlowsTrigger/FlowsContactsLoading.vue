<template>
  <section class="flows-contacts-loading">
    <header>
      <unnnic-skeleton-loading />
      <unnnic-skeleton-loading height="22px" />
    </header>
    <main>
      <ul class="flows-contacts-loading__room-group">
        <li v-for="room in 4" :key="`room-${room}`" class="flows-contacts-loading__room">
          <unnnic-skeleton-loading height="40px" />
          <div class="flows-contacts-loading__room__infos">
            <unnnic-skeleton-loading />
            <unnnic-skeleton-loading />
          </div>
        </li>
      </ul>
    </main>
  </section>
</template>

<script>
export default {
  name: 'FlowsContactsLoading',
};
</script>

<style lang="scss" scoped>
.flows-contacts-loading {
  display: flex;
  flex-direction: column;

  overflow: hidden auto;

  header {
    padding: $unnnic-spacing-ant 0;

    display: grid;
    gap: $unnnic-spacing-xs;
    grid-template-columns: auto 22px;

    & > * {
      width: 100%;
      height: 22px;
    }
  }

  &__room-list {
    margin-bottom: $unnnic-spacing-sm;
  }

  &__room-group {
    padding: 0;
    margin: 0;
  }

  &__room {
    padding: $unnnic-spacing-xs;
    display: grid;
    grid-template-columns: 40px auto 20px;
    gap: $unnnic-spacing-xs;
    align-items: center;

    & > * {
      width: 100%;
      max-height: 40px;
    }

    &__infos {
      display: grid;
      gap: $unnnic-spacing-nano;

      & > * {
        width: 100%;
        height: 16px;
      }
    }
  }
}
</style>
