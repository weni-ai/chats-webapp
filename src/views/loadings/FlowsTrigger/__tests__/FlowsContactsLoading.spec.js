import { mount } from '@vue/test-utils';
import FlowsContactsLoading from '../FlowsContactsLoading.vue';
import { beforeEach, describe, expect } from 'vitest';

const createWrapper = () => {
  return mount(FlowsContactsLoading);
};

describe('FlowsContactsLoading.vue', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = createWrapper();
  });

  it('to match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
