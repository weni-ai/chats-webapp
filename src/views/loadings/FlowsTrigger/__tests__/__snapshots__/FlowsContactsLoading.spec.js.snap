// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FlowsContactsLoading.vue > to match snapshot 1`] = `
"<section data-v-2da6e388="" class="flows-contacts-loading">
  <header data-v-2da6e388=""><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 22px;"> ‌ </span></span></header>
  <main data-v-2da6e388="">
    <ul data-v-2da6e388="" class="flows-contacts-loading__room-group">
      <li data-v-2da6e388="" class="flows-contacts-loading__room"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span>
        <div data-v-2da6e388="" class="flows-contacts-loading__room__infos"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
      </li>
      <li data-v-2da6e388="" class="flows-contacts-loading__room"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span>
        <div data-v-2da6e388="" class="flows-contacts-loading__room__infos"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
      </li>
      <li data-v-2da6e388="" class="flows-contacts-loading__room"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span>
        <div data-v-2da6e388="" class="flows-contacts-loading__room__infos"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
      </li>
      <li data-v-2da6e388="" class="flows-contacts-loading__room"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite; height: 40px;"> ‌ </span></span>
        <div data-v-2da6e388="" class="flows-contacts-loading__room__infos"><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span><span data-v-2da6e388=""><span class="unnnic-skeleton" style="background-color: rgb(232, 244, 244); animation: SkeletonLoading 1.5s ease-in-out infinite;"> ‌ </span></span></div>
      </li>
    </ul>
  </main>
</section>"
`;
